### ISSUE TITLE
Source maps and breakpoints read correctly but app hangs and breakpoint is never hit

### ISSUE THREAD

**Issue by melink14** [microsoft/vscode-js-debug#1164]
**Describe the bug**
I'm running Typescript tests using @web/test-runner with esbuild transpiling the typescript on the fly and providing inline source maps.

 It's a bit strange in that the compiled JS is not on disk so we end up with web `background.ts` mapping to on disk `background.ts`.  Similar set ups have worked in the past so I'm thinking a subtle issue is preventing breakpoints from firing.

(If there are no breakpoints, then the tests run fine and the test which don't contain a breakpoint will also run fine while the final test hangs until I restart)

The diagnostic tool shows everything as expected except that the top level files seem to be absolute paths printed relative to my workspace directory. I assume that's a diagnostic tool only bug since the detailed paths are exactly right.

![image](https://user-images.githubusercontent.com/1176550/147855174-4f1d8b4c-c5cc-499f-8532-79d9f527301a.png)

I'm hoping the log files make the problem obvious but I can push a branch to my repo for easy debugging if needed later.

**Log File**
[vscode-debugadapter-be9fedb0.json.gz](https://github.com/microsoft/vscode-js-debug/files/7797922/vscode-debugadapter-be9fedb0.json.gz)

**VS Code Version:**

> バージョン: 1.63.2 (system setup)
> コミット: 899d46d82c4c95423fb7e10e68eba52050e30ba3
>  日付: 2021-12-15T09:40:02.816Z
> Electron: 13.5.2
> Chromium: 91.0.4472.164
> Node.js: 14.16.0
> V8: 9.1.269.39-electron.0
> OS: Windows_NT x64 10.0.19043

**Additional context**
I'm running remote WSL with debian based linux though it's probably not related.

**Comment by connor4312** [comment]
That all should be fine. It looks like the page crashes for some reason.

```
{
  "method": "Target.targetCrashed",
  "params": {
    "targetId": "1F3BCA1F30D7CC5F868E28D29089320B",
    "status": "crashed",
    "errorCode": 11
  },
  "sessionId": "BF2C5A9D2A6DB08D80CD5C2EF0C24995"
}
```

Unfortunately this indicates an issue in Chrome and is not something I can readily fix in the debugger. I suggest trying to run this with non-headless Chrome to reduce the number of variables going on. If that still crashes, I recommend creating a crash bug for Chromium using the instructions here: https://www.chromium.org/for-testers/bug-reporting-guidelines/reporting-crash-bug

**Comment by melink14** [comment]
Thanks for looking into it connor!

I reproduced with non-headless and I'll look into filing a bug with the crash ID.

I also tried older known working versions of chrome and still got the crash so I guess another variable is causing the problem.

I might also try older versions of the test runner and puppeteer but I wonder if there's a way to try older versions of js-debug as well? I guess if I use an older version of vscode it would work...

**Comment by melink14** [comment]
Another datapoint:
I connected to my test via `chrome://inspect` and was able to set breakpoints and inspect normally (after using a `debugger` statement to access source long enough to set chrome breakpoints) so it seems remote debugging works in general and something about my VS Code setup is breaking it in a way that wasn't breaking it 3-5 months ago...

**Comment by connor4312** [comment]
Sure, you can use any old version of the [nightly build](https://github.com/microsoft/vscode-js-debug#nightly-extension)

**Comment by melink14** [comment]
As an update, I tried the first nightly build from early November with no success and also tried just an old release of my code when I was actively developing.

I wanted to try using the old chrome debugger extension but it didn't seem to work anymore at all.

I guess at this point I can only hope the crash reports reveal something to the chrome engineers. In the mean time, I'll also try on a couple other machines. Maybe also an old release of VS Code would be worth trying too...

**Comment by melink14** [comment]
I got caught up in work but came back to this recently.  I'll continue recording my progress here for future readers.
To reduce variables I removed the realtime transpilation and ran my tests pre-compiled.

I noticed first that, if sourcemaps were off then there was no crash. Then I tried to figure out which source maps caused it by looking at all my deps.

I found that in order for it to not crash in this case, I had to remove the inline sourcemaps of two of my sourcefiles. It has to be both, if i just remove one or the other, the crash still occurs. The odd part is that the breakpoint is not related to these files and so even though the files are loaded via imports analyzing the sourcemaps would not be needed for placing a breakpoint.

Even though the crash is in chrome, I wonder if it could be caused by a malformed command due to an improperly read sourcemap...
